diff --git a/python/getnicinfo.py b/python/getnicinfo.py
index 97afdbc..c6e85f0 100644
--- a/python/getnicinfo.py
+++ b/python/getnicinfo.py
@@ -13,8 +13,16 @@ def GetNicInfo(addr_cmd='ip add ls', route_cmd='ip route ls', hostname_cmd='host
 
     Return a dictionary like:
 
-        nics={"eth0":{"type":"E", "addr":"10.200.1.2/24", "hostname":"toolinux.chaoseternal.net", "route":"10.200.1.254"},
-              "eth1":{"type":"E", "addr":['10.200.1.5/24', '10.200.1.6/24'], "hostname":"yadayadayda", "route":"10.200.1.254"}
+        nics={"eth0":{"type":"E",
+                      "mac":"00:12:AB:CD:EF:AA",
+                      "addr":"10.200.1.2/24",
+                      "hostname":"toolinux.chaoseternal.net",
+                      "route":"10.200.1.254"},
+              "eth1":{"type":"E",
+                      "mac":"00:AA:BB:CC:DD:EE",
+                      "addr":['10.200.1.5/24', '10.200.1.6/24'],
+                      "hostname":"yadayadayda",
+                      "route":"10.200.1.254"}
     """
 
     # get system commands result
@@ -22,14 +30,16 @@ def GetNicInfo(addr_cmd='ip add ls', route_cmd='ip route ls', hostname_cmd='host
     route_raw = os.popen(route_cmd).readlines()
     hostname_raw = os.popen(hostname_cmd).readlines()
 
-    # get nic name, 'type' and 'addr'
+    # get nic name, 'type', 'addr' and 'mac'
     names = [] # ['eth0', 'tun0']
     for name in filter(lambda x:x.find('mtu') >=0 and x.find('lo') ==-1, addr_raw):
         names.append(name.split(':')[1].strip())
     
     types = [] # ['ether', '[65534]']
+    macs = [] # ['00:12:AB:CD:EF:AA', '00:AA:BB:CC:DD:EE']
     for tp in filter(lambda x:x.find('link/') >=0 and x.find('loopback') ==-1, addr_raw):
         types.append(tp.split('/')[1].split(' ')[0])
+        macs.append(tp.split(' ')[5] != '\n' and tp.split(' ')[5]  or 'None')
     nic_type = {'ether':'E', '[65534]':'tun'}
     types = map(lambda x:x in nic_type and x.replace(x, nic_type[x]) or x, types)
 
@@ -51,9 +61,15 @@ def GetNicInfo(addr_cmd='ip add ls', route_cmd='ip route ls', hostname_cmd='host
     routes = filter(lambda x: x.find('default') >= 0, route_raw)
     route = routes[0].split(' ')[2]
 
-    # gather stuff
-    nics = [[names[j], [ ['type', types[j]], ['addr', addrs[j]], ['hostname', hostname], ['route', route] ]] for j in range(len(names))]
-    for i in range(len(names)):
+    # gather stuff, filter no ip addr nics
+    nics = [[names[j], [
+                        ['type', types[j]],
+                        ['mac', macs[j]],
+                        ['addr', addrs[j]],
+                        ['hostname', hostname],
+                        ['route', route] ]
+                       ] for j in range(len(names)) if addrs[j] != []]
+    for i in range(len(nics)):
         nics[i][1] = dict(nics[i][1])
     return dict(nics)
 
